pkg/api/resource/quantity_proto.go

```diff
@@ -17,62 +17,287 @@ limitations under the License.
 package resource
 
 import (
-	"math/big"
+	"fmt"
+	"io"
 
-	inf "gopkg.in/inf.v0"
+	"github.com/gogo/protobuf/proto"
 )
 
-// QuantityProto is a struct that is equivalent to Quantity, but intended for
-// protobuf marshalling/unmarshalling. It is generated into a serialization
-// that matches Quantity. Do not use in Go structs.
-//
-// +protobuf=true
-type QuantityProto struct {
-	// The format of the quantity
-	Format Format `protobuf:"bytes,1,opt,name=format,casttype=Format"`
-	// The scale dimension of the value
-	Scale int32 `protobuf:"varint,2,opt,name=scale"`
-	// Bigint is serialized as a raw bytes array
-	Bigint []byte `protobuf:"bytes,3,opt,name=bigint"`
-}
+var _ proto.Sizer = &Quantity{}
 
-// ProtoTime returns the Time as a new ProtoTime value.
-func (q *Quantity) QuantityProto() *QuantityProto {
-	if q == nil {
-		return &QuantityProto{}
+func (m *Quantity) Marshal() (data []byte, err error) {
```

Comment:
Document the format of the returned data-- I take it it's just a string in the same format as the json output?